_pages/dev/lazy-loading-guide.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Lazy Loading Guide
+---
+
+**Lazy loading or code splitting** lets you divide the JavaScript code into multiple chunks, so they don't have to be loaded upfront, but only when actually needed, greatly improving Time To Interactive, especially in case of complex web applications and on lower-end mobile devices. 
+
+## Spartacus Approach to Lazy Loading
+   
+As Spartacus is mostly CMS driven, there is not much of the usage of the built-in route based lazy loading offered by Angular out of the box. Code splitting has to be done at app build time, while Spartacus never knows upfront what components or features a specific route can contain, before actually loading them. 
+There is always a room for manual tweaking it based on app knowledge, but such a solution breaks the flexibility that Spartacus offers and requires additional prerequisites.
+   
+To meet those requirements, Spartacus provides CMS driven lazy loading on two levels:
+ 
+  1. Lazy loading of CMS components (simplest)
+   	
+  2. CMS driven lazy loading of feature modules
+    
+    
+## Important information
+
+### Dynamic imports can only be defined in the main application
+
+**Dynamic imports**, a technique used to facilitate lazy loading and allow for code splitting, can be only used in the main application. It's not possible to define them in prebuild libraries.
```

Comment:
I'd add something like: 
This is an unfortunate limitation, that results in some application code that must be added by customers. While the amount of custom code is limited to the bare minimum, we'll add a feature in a future version of the schematics library to automatically add lazy loaded modules.